packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -195,6 +195,9 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
       }
 
       if (this._connected) {
+        // if the selected wallet is already connected, we don't need to connect again
+        if (this.wallet.name === walletName) return;
```

Comment:
lets return a message saying `${walletName} is already connected` or something like that, so dapps can let their users know why nothing is happening

Reply:
return or log this?

Reply:
return, so wallets can display it to the end user however they want. something like we do [here](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-core/src/WalletCore.ts#L406)

Reply:
ok, I added some result structure and enum